### PR TITLE
remove tooltip text in expert filter components

### PR DESCRIPTION
**Pull request by basseche**

### CODE REVIEW

**Review by Mathieu-Deharbe — CHANGES_REQUESTED** [review]
There is one left : 

tooltips

I think the second one "Rule group at path 2" should be removed as well. It is part of the react query builder auto tooltip system. But see with the PO.

Plus (if you feel like it) "Drag handle" but I am not sure at all about it.

**Comment by basseche** [comment]
> There is one left :
> 
> tooltips [ tooltips ](https://private-user-images.githubusercontent.com/148252167/375436591-a5ac2392-6aa1-4d0c-90bc-b1aed3256ce1.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg5MDU4NTQsIm5iZiI6MTcyODkwNTU1NCwicGF0aCI6Ii8xNDgyNTIxNjcvMzc1NDM2NTkxLWE1YWMyMzkyLTZhYTEtNGQwYy05MGJjLWIxYWVkMzI1NmNlMS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxNFQxMTMyMzRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zYWU2NjIzMTJlODk4ZmI0ZWJlMzZlNjZhNGZkNDFjY2U1MGZiNWU2MmNkY2NlMmZiM2E4NWY0NmY4NzA5OWVjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.d8O_E3g74hIcQ6Db9HLA700-Loo75ekmJ1Snf8mUJNE) [ ](https://private-user-images.githubusercontent.com/148252167/375436591-a5ac2392-6aa1-4d0c-90bc-b1aed3256ce1.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg5MDU4NTQsIm5iZiI6MTcyODkwNTU1NCwicGF0aCI6Ii8xNDgyNTIxNjcvMzc1NDM2NTkxLWE1YWMyMzkyLTZhYTEtNGQwYy05MGJjLWIxYWVkMzI1NmNlMS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxNFQxMTMyMzRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zYWU2NjIzMTJlODk4ZmI0ZWJlMzZlNjZhNGZkNDFjY2U1MGZiNWU2MmNkY2NlMmZiM2E4NWY0NmY4NzA5OWVjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.d8O_E3g74hIcQ6Db9HLA700-Loo75ekmJ1Snf8mUJNE)
> 
> I think the second one "Rule group at path 2" should be removed as well. It is part of the react query builder auto tooltip system. But see with the PO.
> 
> Plus (if you feel like it) "Drag handle" but I am not sure at all about it.

@Mathieu-Deharbe : all resolved in new commit

**Review by Mathieu-Deharbe — APPROVED** [review]
Found a last kind of useless tooltip. Not asked in the initial ticket but do as you wish : 

open tooltip

Approved anyway.

File: src/components/inputs/reactQueryBuilder/CustomReactQueryBuilder.tsx [hunk 1]
```diff
@@ -75,6 +82,10 @@ export function CustomReactQueryBuilder(props: Readonly<CustomReactQueryBuilderP
         }));
     }, [intl]);
 
+    // Drag handle title
+    const customDragHandle = { label: defaultTranslations.dragHandle.label, title: '' };
+    const customTranslations = { ...defaultTranslations, dragHandle: customDragHandle };
+
```
Comment (thangqp) [inline src/components/inputs/reactQueryBuilder/CustomReactQueryBuilder.tsx:88]: We can use useMemo to override the whole defaultTranslations with empty title for components.
In the future, if we need translation, only replace empties by useInlt

File: src/components/inputs/reactQueryBuilder/CombinatorSelector.tsx [hunk 2]
```diff
@@ -31,7 +31,7 @@ export function CombinatorSelector(props: CombinatorSelectorProps) {
             />
             <MaterialValueSelector
                 {...props}
-                title={undefined} // disable the tooltip
+                title="" // disable the tooltip
```
Comment (thangqp) [inline src/components/inputs/reactQueryBuilder/CombinatorSelector.tsx:34]: To remove

File: src/components/inputs/reactQueryBuilder/CustomReactQueryBuilder.tsx [hunk 3]
```diff
@@ -32,11 +39,11 @@ export interface CustomReactQueryBuilderProps {
 }
 
 function RuleAddButton(props: Readonly<ActionWithRulesAndAddersProps>) {
-    return <AddButton {...props} label="rule" />;
+    return <AddButton {...props} label="rule" title="" />;
```
Comment (thangqp) [inline src/components/inputs/reactQueryBuilder/CustomReactQueryBuilder.tsx:42]:
```suggestion
    return <AddButton {...props} label="rule" />;
```

File: src/components/inputs/reactQueryBuilder/CustomReactQueryBuilder.tsx [hunk 4]
```diff
@@ -32,11 +39,11 @@ export interface CustomReactQueryBuilderProps {
...
 }
 
 function GroupAddButton(props: Readonly<ActionWithRulesAndAddersProps>) {
-    return <AddButton {...props} label="subGroup" />;
+    return <AddButton {...props} label="subGroup" title="" />;
```
Comment (thangqp) [inline src/components/inputs/reactQueryBuilder/CustomReactQueryBuilder.tsx:46]:
```suggestion
    return <AddButton {...props} label="subGroup" />;
```

File: src/components/inputs/reactQueryBuilder/FieldSelector.tsx [hunk 5]
```diff
@@ -24,6 +24,7 @@ export function FieldSelector({ options, className, value, disabled, handleOnCha
             options={optionList}
             disableClearable
             size="small"
+            title=""
```
Comment (thangqp) [inline src/components/inputs/reactQueryBuilder/FieldSelector.tsx:27]:
```suggestion
            title={title}
```

File: src/components/inputs/reactQueryBuilder/RemoveButton.tsx [hunk 6]
```diff
@@ -29,7 +29,7 @@ export function RemoveButton(props: ActionWithRulesProps) {
     const isLastRuleOrGroup = path.toString() === [0].toString() && getNumberOfSiblings(path, query) === 1;
 
     return (
-        <IconButton size="small" onClick={() => handleDelete()} className={className}>
+        <IconButton size="small" onClick={() => handleDelete()} className={className} title="">
```
Comment (thangqp) [inline src/components/inputs/reactQueryBuilder/RemoveButton.tsx:32]:
```suggestion
        <IconButton size="small" onClick={() => handleDelete()} className={className} title={title}>
```

File: src/components/inputs/reactQueryBuilder/TextValueEditor.tsx [hunk 7]
```diff
@@ -16,15 +16,10 @@ export function TextValueEditor(props: ValueEditorProps) {
...
+    const { value, handleOnChange, title } = props;
     // The displayed component totally depends on the value type and not the operator. This way, we have smoother transition.
     if (!Array.isArray(value)) {
-        return (
-            <MaterialValueEditor
-                {...props}
-                title={undefined} // disable the tooltip
-            />
-        );
+        return <MaterialValueEditor {...props} title={title} />;
```
Comment (thangqp) [inline src/components/inputs/reactQueryBuilder/TextValueEditor.tsx:22]:
```suggestion
        return <MaterialValueEditor {...props} />;
```

File: src/components/inputs/reactQueryBuilder/TranslatedValueEditor.tsx [hunk 8]
```diff
@@ -39,18 +39,13 @@ export function TranslatedValueEditor(props: ValueEditorProps) {
...
     // The displayed component totally depends on the value type and not the operator. This way, we have smoother transition.
     if (!Array.isArray(value)) {
-        return (
-            <MaterialValueEditor
-                {...props}
-                values={translatedValues}
-                title={undefined} // disable the tooltip
-            />
-        );
+        return <MaterialValueEditor {...props} values={translatedValues} title={title} />;
```
Comment (thangqp) [inline src/components/inputs/reactQueryBuilder/TranslatedValueEditor.tsx:42]:
```suggestion
        return <MaterialValueEditor {...props} values={translatedValues} />;
```

File: src/components/inputs/reactQueryBuilder/ValueEditor.tsx [hunk 9]
```diff
@@ -66,11 +66,11 @@ export function ValueEditor(props: ValueEditorProps) {
     }
 
     if ([FieldType.COUNTRY, FieldType.COUNTRY_1, FieldType.COUNTRY_2].includes(field as FieldType)) {
-        return <CountryValueEditor {...props} />;
+        return <CountryValueEditor {...props} title="" />;
```
Comment (thangqp) [inline src/components/inputs/reactQueryBuilder/ValueEditor.tsx:69]:
```suggestion
        return <CountryValueEditor {...props} />;
```

File: src/components/inputs/reactQueryBuilder/ValueEditor.tsx [hunk 10]
```diff
@@ -66,11 +66,11 @@ export function ValueEditor(props: ValueEditorProps) {
...
     }
 
     if (fieldData.dataType === DataType.ENUM) {
-        return <TranslatedValueEditor {...props} />;
+        return <TranslatedValueEditor {...props} title="" />;
```
Comment (thangqp) [inline src/components/inputs/reactQueryBuilder/ValueEditor.tsx:73]:
```suggestion
        return <TranslatedValueEditor {...props} />;
```

File: src/components/inputs/reactQueryBuilder/ValueEditor.tsx [hunk 11]
```diff
@@ -103,7 +103,7 @@ export function ValueEditor(props: ValueEditorProps) {
     }
 
     if (fieldData.dataType === DataType.STRING) {
-        return <TextValueEditor {...props} />;
+        return <TextValueEditor {...props} title="" />;
```
Comment (thangqp) [inline src/components/inputs/reactQueryBuilder/ValueEditor.tsx:106]:
```suggestion
        return <TextValueEditor {...props} />;
```

File: src/components/inputs/reactQueryBuilder/PropertyValueEditor.tsx [hunk 12]
```diff
@@ -22,10 +22,11 @@ const PROPERTY_VALUE_OPERATORS = [OPERATOR_OPTIONS.IN];
 interface ExpertFilterPropertyProps {
     equipmentType: EquipmentType;
     valueEditorProps: ValueEditorProps;
+    title: string;
```
Comment (thangqp) [inline src/components/inputs/reactQueryBuilder/PropertyValueEditor.tsx:25]: To remove

File: src/components/inputs/reactQueryBuilder/PropertyValueEditor.tsx [hunk 13]
```diff
@@ -84,12 +85,14 @@ export function PropertyValueEditor(props: ExpertFilterPropertyProps) {
                         onChange(FieldConstants.PROPERTY_NAME, value);
                     }}
                     size="small"
+                    title={title}
```
Comment (thangqp) [inline src/components/inputs/reactQueryBuilder/PropertyValueEditor.tsx:88]:
```suggestion
                    title={valueEditorProp?.title}
```

File: src/components/inputs/reactQueryBuilder/PropertyValueEditor.tsx [hunk 14]
```diff
@@ -84,12 +85,14 @@ export function PropertyValueEditor(props: ExpertFilterPropertyProps) {
...
                 />
             </Grid>
             <Grid item xs={2.5}>
                 <Select
                     value={propertyOperator ?? PROPERTY_VALUE_OPERATORS[0].customName}
                     size="small"
+                    title={title}
```
Comment (thangqp) [inline src/components/inputs/reactQueryBuilder/PropertyValueEditor.tsx:95]:
```suggestion
                    title={valueEditorProp?.title}
```

File: src/components/inputs/reactQueryBuilder/PropertyValueEditor.tsx [hunk 15]
```diff
@@ -106,6 +109,7 @@ export function PropertyValueEditor(props: ExpertFilterPropertyProps) {
                 <Autocomplete
                     value={propertyValues ?? []}
                     options={predefinedValues ?? []}
+                    title={title}
```
Comment (thangqp) [inline src/components/inputs/reactQueryBuilder/PropertyValueEditor.tsx:112]:
```suggestion
                    title={valueEditorProp?.title}
```

File: src/components/inputs/reactQueryBuilder/ValueEditor.tsx [hunk 16]
```diff
@@ -126,18 +126,18 @@ export function ValueEditor(props: ValueEditorProps) {
             equipmentType = getValues(FieldConstants.EQUIPMENT_TYPE);
         }
 
-        return <PropertyValueEditor equipmentType={equipmentType} valueEditorProps={props} />;
+        return <PropertyValueEditor equipmentType={equipmentType} valueEditorProps={props} title="" />;
```
Comment (thangqp) [inline src/components/inputs/reactQueryBuilder/ValueEditor.tsx:129]:
```suggestion
        return <PropertyValueEditor equipmentType={equipmentType} valueEditorProps={props} />;
```

File: src/components/inputs/reactQueryBuilder/ValueEditor.tsx [hunk 17]
```diff
@@ -126,18 +126,18 @@ export function ValueEditor(props: ValueEditorProps) {
...
     }
 
     if (fieldData.dataType === DataType.COMBINATOR) {
-        return <GroupValueEditor {...props} />;
+        return <GroupValueEditor {...props} title="" />;
```
Comment (thangqp) [inline src/components/inputs/reactQueryBuilder/ValueEditor.tsx:133]:
```suggestion
        return <GroupValueEditor {...props} />;
```

File: src/components/inputs/reactQueryBuilder/ValueEditor.tsx [hunk 18]
```diff
@@ -126,18 +126,18 @@ export function ValueEditor(props: ValueEditorProps) {
...
     }
 
     return (
         <Box sx={inputType === 'number' ? styles.noArrows : undefined}>
             <MaterialValueEditor
                 {...props}
-                title={undefined} // disable the tooltip
+                title="" // disable the tooltip
```
Comment (thangqp) [inline src/components/inputs/reactQueryBuilder/ValueEditor.tsx:140]: To remove

File: src/components/inputs/reactQueryBuilder/ValueSelector.tsx [hunk 19]
```diff
@@ -12,7 +12,7 @@ export function ValueSelector(props: ValueSelectorProps) {
     return (
         <MaterialValueSelector
             {...props}
-            title={undefined} // disable the tooltip
+            title="" // disable the tooltip
```
Comment (thangqp) [inline src/components/inputs/reactQueryBuilder/ValueSelector.tsx:15]: To remove

File: src/components/inputs/reactQueryBuilder/CustomReactQueryBuilder.tsx (hunk 4 above)
Comment (basseche) [reply src/components/inputs/reactQueryBuilder/CustomReactQueryBuilder.tsx:46]: ok thanks

File: src/components/inputs/reactQueryBuilder/CombinatorSelector.tsx (hunk 2 above)
Comment (basseche) [reply src/components/inputs/reactQueryBuilder/CombinatorSelector.tsx:34]: ok

File: src/components/inputs/reactQueryBuilder/CustomReactQueryBuilder.tsx (hunk 3 above)
Comment (basseche) [reply src/components/inputs/reactQueryBuilder/CustomReactQueryBuilder.tsx:42]: ok

File: src/components/inputs/reactQueryBuilder/FieldSelector.tsx (hunk 5 above)
Comment (basseche) [reply src/components/inputs/reactQueryBuilder/FieldSelector.tsx:27]: ok

File: src/components/inputs/reactQueryBuilder/TranslatedValueEditor.tsx (hunk 8 above)
Comment (basseche) [reply src/components/inputs/reactQueryBuilder/TranslatedValueEditor.tsx:42]: ok

File: src/components/inputs/reactQueryBuilder/CustomReactQueryBuilder.tsx [hunk 20]
```diff
@@ -39,13 +39,27 @@ export interface CustomReactQueryBuilderProps {
...
+    fields: { ...defaultTranslations.fields, title: '' },
+    operators: { ...defaultTranslations.operators, title: '' },
+    dragHandle: { ...defaultTranslations.dragHandle, title: '' },
+    addRule: { ...defaultTranslations.addRule, title: '' },
+    addGroup: { ...defaultTranslations.addGroup, title: '' },
+    removeRule: { ...defaultTranslations.removeRule, title: '' },
+    removeGroup: { ...defaultTranslations.removeGroup, title: '' },
+    value: { title: '' },
+    combinators: { title: '' },
+};
```
Comment (Mathieu-Deharbe) [inline src/components/inputs/reactQueryBuilder/CustomReactQueryBuilder.tsx:61]: Nice ! Ok @thangqp you are and remain the React query builder master.

**Review by Mathieu-Deharbe — APPROVED** [review]

**Review by thangqp — APPROVED** [review]
Test OK

**Pull request merged**